server/datastore/mysql/microsoft_mdm.go

```diff
@@ -135,136 +117,114 @@ func (ds *Datastore) MDMWindowsDeleteEnrolledDeviceWithDeviceID(ctx context.Cont
+		}
+
+		// create the command execution queue entries, one per host
+		for _, hostUUID := range hostUUIDs {
+			if err := ds.mdmWindowsInsertHostCommandDB(ctx, tx, hostUUID, cmd.CommandUUID); err != nil {
 				return err
 			}
 		}
 		return nil
 	})
 }
 
-// MDMWindowsInsertPendingCommand inserts a new WindowMDMPendingCommand in the database
-func (ds *Datastore) MDMWindowsInsertPendingCommand(ctx context.Context, cmd *fleet.MDMWindowsPendingCommand) error {
-	return ds.mdmWindowsInsertPendingCommandDB(ctx, ds.writer(ctx), cmd)
-}
-
-func (ds *Datastore) mdmWindowsInsertPendingCommandDB(ctx context.Context, tx sqlx.ExecerContext, cmd *fleet.MDMWindowsPendingCommand) error {
+func (ds *Datastore) mdmWindowsInsertHostCommandDB(ctx context.Context, tx sqlx.ExecerContext, hostUUID, commandUUID string) error {
 	stmt := `
-		INSERT INTO old_windows_mdm_pending_commands (
+	INSERT INTO windows_mdm_command_queue (
+		enrollment_id,
 		command_uuid,
-		device_id,
-		cmd_verb,
-		setting_uri,
-		setting_value,
-		data_type,
-		system_origin ) VALUES (?, ?, ?, ?, ?, ?, ?)
-	`
-	_, err := tx.ExecContext(
-		ctx,
-		stmt,
-		cmd.CommandUUID,
-		cmd.DeviceID,
-		cmd.CmdVerb,
-		cmd.SettingURI,
-		cmd.SettingValue,
-		cmd.DataType,
-		cmd.SystemOrigin)
-	if err != nil {
+	)
+	VALUES
+		(
+			SELECT
+				id, ?
+			FROM
+				mdm_windows_enrollments
+			WHERE
+				host_uuid = ?
+		)
+`
+	if _, err := tx.ExecContext(ctx, stmt, commandUUID, hostUUID); err != nil {
 		if isDuplicate(err) {
-			return ctxerr.Wrap(ctx, alreadyExists("MDMWindowsPendingCommand", cmd.CommandUUID))
+			return ctxerr.Wrap(ctx, alreadyExists("MDMWindowsCommandQueue", commandUUID))
 		}
-		return ctxerr.Wrap(ctx, err, "inserting MDMWindowsPendingCommand")
+		return ctxerr.Wrap(ctx, err, "inserting MDMWindowsCommandQueue")
 	}
 
 	return nil
 }
 
-// MDMWindowsGetPendingCommands retrieves all commands for a given device ID from the windows_mdm_pending_commands table
-func (ds *Datastore) MDMWindowsGetPendingCommands(ctx context.Context, deviceID string) ([]*fleet.MDMWindowsPendingCommand, error) {
-	var commands []*fleet.MDMWindowsPendingCommand
+// MDMWindowsGetPendingCommands retrieves all commands awaiting execution for a
+// given device ID.
+func (ds *Datastore) MDMWindowsGetPendingCommands(ctx context.Context, deviceID string) ([]*fleet.MDMWindowsCommand, error) {
+	var commands []*fleet.MDMWindowsCommand
 
 	query := `
-        SELECT
-            command_uuid,
-            device_id,
-            cmd_verb,
-            setting_uri,
-            setting_value,
-            data_type,
-            system_origin,
-            created_at,
-            updated_at
-        FROM
-            old_windows_mdm_pending_commands wmpc
-        WHERE
-            wmpc.device_id = ? AND
-            NOT EXISTS (SELECT 1 FROM old_windows_mdm_commands wmc WHERE wmpc.device_id = wmc.device_id AND wmpc.command_uuid = wmc.command_uuid)
-    `
+SELECT
```

Comment:
Note that while I adjusted the query, it is still untested and might not work as expected.